common/persistence/data_store_interfaces.go

```diff
@@ -922,8 +922,9 @@ func NewDataBlob(data []byte, encodingType common.EncodingType) *DataBlob {
 	if len(data) == 0 {
 		return nil
 	}
-	if encodingType != "thriftrw" && data[0] == 'Y' {
-		panic(fmt.Sprintf("Invalid incoding: \"%v\"", encodingType))
+	if encodingType != common.EncodingTypeThriftRW && data[0] == 'Y' {
+		// original reason for this is not written down, but maybe for handling data prior to an encoding type?
+		panic(fmt.Sprintf("Invalid data blob encoding: \"%v\"", encodingType))
```

Comment:
There seems no reason. Temporal removed that logic: https://github.com/temporalio/temporal/pull/638/files#diff-f4778e6c9f74affb5767ad2ce6672a85ac6ba16eb64b42057e7dcef273ed5ac0L641

Reply:
we should probably remove it at some point then too.
I'll leave that for another diff / let's make a task or something around it, and see if thrift/proto ever start with Y because what even